### PR TITLE
Fix empty string check inconsistency in validation.go

### PR DESCRIPTION
**Pull request by resouer**
As the title said, and cc @pmorie  as we talked this before.

### CODE REVIEW

**Comment by feihujiang** [comment]
:+1: LGTM

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -1351,7 +1351,7 @@ func ValidateService(service *api.Service) validation.ErrorList {
 func validateServicePort(sp *api.ServicePort, requireName, isHeadlessService bool, allNames *sets.String, fldPath *validation.FieldPath) validation.ErrorList {
 	allErrs := validation.ErrorList{}
 
-	if requireName && sp.Name == "" {
+	if requireName && len(sp.Name) == 0 {
 		allErrs = append(allErrs, validation.NewRequiredError(fldPath.Child("name")))
 	} else if sp.Name != "" {
```
Comment (pmorie) [inline pkg/api/validation/validation.go]: What about these?

Comment (resouer) [inline pkg/api/validation/validation.go]: A fish escaping from the net :)

**Comment by resouer** [comment]
@k8s-bot test this

**Comment by resouer** [comment]
@pmorie Needs a lgtm :)

**Comment by resouer** [comment]
what happened ... @k8s-bot test this please

**Pull request merged**